shooter_rocket/init.lua

```diff
@@ -70,33 +81,43 @@ minetest.register_tool("shooter_rocket:rocket_gun_loaded", {
 	inventory_image = "shooter_rocket_gun_loaded.png",
 	groups = {not_in_creative_inventory=1},
 	on_use = function(itemstack, user, pointed_thing)
-		if not minetest.setting_getbool("creative_mode") then
-			itemstack:add_wear(65535 / 50)
-		end
-		itemstack = "shooter_rocket:rocket_gun 1 "..itemstack:get_wear()
-		if pointed_thing.type ~= "nothing" then
-			local pointed = minetest.get_pointed_thing_position(pointed_thing)
-			if vector.distance(user:get_pos(), pointed) < 8 then
-				shooter.blast(pointed, 2, 50, 7)
-				return itemstack
+		if plcooldown[user:get_player_name()] ~= 0 then
+			minetest.chat_send_player(user:get_player_name(), "Your rocket has a cooldown!")
+		else
+			if not minetest.setting_getbool("creative_mode") then
```

Comment:
The method you're using is deprecated. Use this one instead.
```suggestion
			if not minetest.settings:get_bool("creative_mode") then
```